olp-cpp-sdk-authentication/src/AuthenticationClientUtils.cpp

```diff
@@ -262,6 +266,54 @@ AuthorizeResult GetAuthorizeResult(rapidjson::Document& doc) {
   return result;
 }
 
+UserAccountInfoResponse GetUserAccountInfoResponse(rapidjson::Document& doc) {
+  const NameToMethodVector<UserAccountInfoResponse, std::string> string_fields{
+      {Constants::USER_ID, &UserAccountInfoResponse::SetUserId},
+      {Constants::REALM, &UserAccountInfoResponse::SetRealm},
+      {Constants::FACEBOOK_ID, &UserAccountInfoResponse::SetFacebookId},
+      {Constants::FIRSTNAME, &UserAccountInfoResponse::SetFirstname},
+      {Constants::LASTNAME, &UserAccountInfoResponse::SetLastname},
+      {Constants::EMAIL, &UserAccountInfoResponse::SetEmail},
+      {Constants::RECOVERY_EMAIL, &UserAccountInfoResponse::SetRecoveryEmail},
+      {Constants::DOB, &UserAccountInfoResponse::SetDob},
+      {Constants::COUNTRY_CODE, &UserAccountInfoResponse::SetCountryCode},
+      {Constants::LANGUAGE, &UserAccountInfoResponse::SetLanguage},
+      {Constants::PHONE_NUMBER, &UserAccountInfoResponse::SetPhoneNumber},
+      {Constants::STATE, &UserAccountInfoResponse::SetState},
+      {Constants::HRN, &UserAccountInfoResponse::SetHrn},
+      {Constants::ACCOUNT_TYPE, &UserAccountInfoResponse::SetAccountType}};
+
+  const NameToMethodVector<UserAccountInfoResponse, bool> boolean_fields{
+      {Constants::EMAIL_VERIFIED, &UserAccountInfoResponse::SetEmailVerified},
+      {Constants::PHONE_NUMBER_VERIFIED,
+       &UserAccountInfoResponse::SetPhoneNumberVerified},
+      {Constants::MARKETING_ENABLED,
+       &UserAccountInfoResponse::SetMarketingEnabled}};
+
+  const NameToMethodVector<UserAccountInfoResponse, time_t> time_fields{
+      {Constants::CREATED_TIME, &UserAccountInfoResponse::SetCreatedTime},
+      {Constants::UPDATED_TIME, &UserAccountInfoResponse::SetUpdatedTime}};
+
+  UserAccountInfoResponse response;
+
+  for (const auto& string_field : string_fields) {
+    if (doc.HasMember(string_field.first) && doc[string_field.first].IsString())
+      (response.*string_field.second)(doc[string_field.first].GetString());
+  }
+
+  for (const auto& boolean_field : boolean_fields) {
+    if (doc.HasMember(boolean_field.first) && doc[boolean_field.first].IsBool())
+      (response.*boolean_field.second)(doc[boolean_field.first].GetBool());
+  }
+
+  for (const auto& time_field : time_fields) {
+    if (doc.HasMember(time_field.first) && doc[time_field.first].IsInt64())
+      (response.*time_field.second)(doc[time_field.first].GetInt64());
+  }
+
+  return response;
```

Comment:
Can we print a warning if there is some field that we don't expect?

Reply:
Yes, we can. But I want to mention that checks about a type, about unexpected fields, etc, will be present only here. For me it's strange to see one fortified method and a dozen methods with possible crashes on each 'Get*'. I mean that we need to refactor all those unsafe functions